src/MCMCTempering.jl

```diff
@@ -39,16 +42,80 @@ implements_logdensity(x) = LogDensityProblems.capabilities(x) !== nothing
 maybe_wrap_model(model) = implements_logdensity(model) ? AbstractMCMC.LogDensityModel(model) : model
 maybe_wrap_model(model::AbstractMCMC.LogDensityModel) = model
 
+# Bundling.
+# TODO: Improve this, somehow.
+# TODO: Move this to an extension.
 function AbstractMCMC.bundle_samples(
-    ts::AbstractVector,
+    ts::AbstractVector{<:TemperedTransition},
     model::AbstractMCMC.AbstractModel,
     sampler::TemperedSampler,
     state::TemperedState,
```

Comment:
```suggestion
    state,
```

I think we should drop the type signature on state, as it means a user needs to somehow access the states to call `bundle_samples` manually, this was in my other PR

Reply:
But we _are_ using methods that expects this type, no? 
The reason behind being restrictive with types here are:
1. `bundle_samples` should really just be compatible with whatever the `step` can return for that particular combination of inputs, hence it makes sense to specialize on the same types as we do in `step`.
2. Not specializing on the `state` here can potentially lead to unnecessary method ambiguities since this is heavily overloaded method.